src/global_heap.cc

```diff
@@ -86,7 +86,11 @@ void GlobalHeap::freeFor(MiniHeap *mh, void *ptr, size_t startEpoch) {
   // read inUseCount before calling free to avoid stalling after the
   // LOCK CMPXCHG in mh->free
   auto remaining = mh->inUseCount() - 1;
-  mh->free(arenaBegin(), ptr);
+
+  // here can't call mh->free(arenaBegin(), ptr), because in consume takeBitmap always clear the bitmap, 
+  // if clearIfNotFree after takeBitmap
+  // it alwasy return false, but in this case, you need to free again.
+  auto wasSet = mh->clearIfNotFree(arenaBegin(), ptr);
```

Comment:
I think I follow - this will return a boolean about whether the race to clear the bit was won or lost against `takeBitmap`, is that right?

Reply:
Yes. I just follow your pre-version, may here you should change a better function name or make the mh->free() can return a bool.